bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaWaterHeatingSystemHandler.java

```diff
@@ -0,0 +1,220 @@
+
+import static org.openhab.binding.somfytahoma.internal.SomfyTahomaBindingConstants.*;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.somfytahoma.internal.model.SomfyTahomaState;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SomfyTahomaWaterHeatingSystemHandler} is responsible for handling commands,
+ * which are sent to one of the channels of the Water Heating system thing.
+ *
+ * @author Benjamin Lafois - Initial contribution
+ */
+@NonNullByDefault
+public class SomfyTahomaWaterHeatingSystemHandler extends SomfyTahomaBaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SomfyTahomaWaterHeatingSystemHandler.class);
+
+    private boolean boostMode = false;
+    private boolean awayMode = false;
+
+    public SomfyTahomaWaterHeatingSystemHandler(Thing thing) {
+        super(thing);
+
+        stateNames.put(MANUFACTURER, MANUFACTURER_NAME_STATE);
+        stateNames.put(MIDDLEWATER_TEMPERATURE, MIDDLE_WATER_TEMPERATURE_STATE);
+        stateNames.put(TARGET_TEMPERATURE, TARGET_TEMPERATURE_STATE);
+        // stateNames.put(WATER_TEMPERATURE, TEMPERATURE_STATE);
+        stateNames.put(WATER_HEATER_MODE, WATER_HEATER_MODE_STATE);
+
+        stateNames.put(BOOST_MODE_DURATION, BOOST_MODE_DURATION_STATE);
+        // override state type because the cloud sends consumption in percent
+        cacheStateType(BOOST_MODE_DURATION_STATE, TYPE_DECIMAL);
+
+        stateNames.put(AWAY_MODE_DURATION, AWAY_MODE_DURATION_STATE);
+        // override state type because the cloud sends consumption in percent
+        cacheStateType(AWAY_MODE_DURATION_STATE, TYPE_DECIMAL);
+
+        stateNames.put(HEAT_PUMP_OPERATING_TIME, HEAT_PUMP_OPERATING_TIME_STATE);
+        // override state type because the cloud sends consumption in percent
+        cacheStateType(HEAT_PUMP_OPERATING_TIME_STATE, TYPE_DECIMAL);
+
+        stateNames.put(ELECTRIC_BOOSTER_OPERATING_TIME, ELECTRIC_BOOSTER_OPERATING_TIME_STATE);
+        // override state type because the cloud sends consumption in percent
+        cacheStateType(ELECTRIC_BOOSTER_OPERATING_TIME_STATE, TYPE_DECIMAL);
+
+        stateNames.put(POWER_HEAT_PUMP, POWER_HEAT_PUMP_STATE);
+        // override state type because the cloud sends consumption in percent
+        cacheStateType(POWER_HEAT_PUMP_STATE, TYPE_DECIMAL);
+
+        stateNames.put(POWER_HEAT_ELEC, POWER_HEAT_ELEC_STATE);
+        // override state type because the cloud sends consumption in percent
+        cacheStateType(POWER_HEAT_ELEC_STATE, TYPE_DECIMAL);
```

Comment:
Is it really necessary to override percent values to decimal? I used this only rare occasions when the cloud was sending unconsistent responses (a bug in the API)

Reply:
Yes - the API is incorrectly returning them as percent - so I had to override it manually